src/effects/effectchainslot.cpp

```diff
@@ -192,37 +192,21 @@ void EffectChainSlot::setDescription(const QString& description) {
 }
 
 void EffectChainSlot::loadEffect(const unsigned int iEffectSlotNumber,
-                                 const QString& id) {
+        const QString& id, EffectManifestPointer pManifest,
+        EffectInstantiatorPointer pInstantiator) {
     EffectSlotPointer pEffectSlot = getEffectSlot(iEffectSlotNumber);
 
-    bool loadNew = false;
-    if (pEffectSlot == nullptr || pEffectSlot->getManifest() == nullptr) {
-        loadNew = true;
-    } else if (id != pEffectSlot->getManifest()->id()) {
-        loadNew = true;
+    if (pEffectSlot == nullptr) {
+        return;
     }
 
-    if (loadNew) {
-        VERIFY_OR_DEBUG_ASSERT(m_pEffectsManager->instantiateEffect(id, pEffectSlot, m_enabledInputChannels)) {
-            qWarning() << "Unable to load effect.";
+    if (pEffectSlot->getManifest() == nullptr || pEffectSlot->getManifest()->id() != id) {
+        VERIFY_OR_DEBUG_ASSERT(pEffectSlot->loadEffect(pManifest, pInstantiator, m_enabledInputChannels)) {
```

Comment:
Remove this VERIFY_OR_DEBUG_ASSERT and make EffectSlot::loadEffect return void. Any errors that happen when EffectSlot::loadEffect is called should be handled by VERIFY_OR_DEBUG_ASSERTs in that function.